src/audio/module_adapter/module/volume/volume.c

```diff
@@ -252,7 +252,10 @@ static inline int32_t volume_windows_fade_ramp(struct vol_data *cd, int32_t ramp
 	int32_t pow_value; /* Q2.30 */
 	int32_t volume_delta = cd->tvolume[channel] - cd->rvolume[channel]; /* Q16.16 */
 
-	time_ratio = (((int64_t)ramp_time) << 30) / (cd->initial_ramp << 3);
+	if (cd->initial_ramp != 0)
+		time_ratio = (((int64_t)ramp_time) << 30) / (cd->initial_ramp << 3);
+	else
+		time_ratio = (((int64_t)ramp_time) << 30);
```

Comment:
in case driver configures ramp time to 0, it does not make sense to enter these functions at all. its hard to predict what will be the result of continuing these volume calculation with non-natural time_ratio.
I think cd->initial_ramp should be checked to be non-zero in volume_process(), otherwise do not enter volume_ramp()

Reply:
or maybe even better solution: in init_ramp(): if (!cd->initial_ramp): cd->ramp_finished = true
Then volume_process will not enter ramp function

Reply:
,,or maybe even better solution: in init_ramp(): if (!cd->initial_ramp): cd->ramp_finished = true
Then volume_process will not enter ramp function" - After this correction, we enter to the ramp function and division by zero continues to occur.

in case driver configures ramp time to 0, it does not make sense to enter these functions at all. its hard to predict what will be the result of continuing these volume calculation with non-natural time_ratio.
I think cd->initial_ramp should be checked to be non-zero in volume_process(), otherwise do not enter volume_ramp()- signal verification does not pass after this amendment.

Reply:
Done